qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -150,30 +124,58 @@ def run(self, dag):
         Returns:
             DAGCircuit: the optimized DAG.
         """
-        if self._decomposers is None:
-            logger.info("Skipping pass because no basis is set")
+        if self._basis_gates is None and self._target is None:
+            logger.info("Skipping pass because no basis or target is set")
             return dag
 
         runs = dag.collect_1q_runs()
         for run in runs:
-            # SPECIAL CASE: Don't bother to optimize single U3 gates which are in the basis set.
-            #     The U3 decomposer is only going to emit a sequence of length 1 anyhow.
-            if "u3" in self._target_basis and len(run) == 1 and isinstance(run[0].op, U3Gate):
-                # Toss U3 gates equivalent to the identity; there we get off easy.
-                if np.allclose(run[0].op.to_matrix(), np.eye(2), 1e-15, 0):
-                    dag.remove_op_node(run[0])
-                    continue
-                # We might rewrite into lower `u`s if they're available.
-                if "u2" not in self._target_basis and "u1" not in self._target_basis:
-                    continue
-
-            new_basis, new_circ = self._resynthesize_run(run)
-
-            if new_circ is not None and self._substitution_checks(dag, run, new_circ, new_basis):
+            qubit = dag.qubits.index(run[0].qargs[0])
```

Comment:
This will get expensive over time because each loop iteration it's an `O(n)` lookup for the index. Typically the pattern we've used to avoid this to just build a mapping outside the for loop like:

```
qubit_indices = {bit: index for index, bit in enumerate(dag.qubits)
```

and then just change this line to be: `qubit = qubit_indices[run[0].qargs[0]]`

Reply:
thanks for the tip, fixed in https://github.com/Qiskit/qiskit-terra/pull/8917/commits/f26ccdfd6bbd42fde50678ee629facb4b92b5b68